### PR TITLE
Warn when an MSI or cabinet installed by an MSI is too large

### PR DESCRIPTION
**Pull request by robmen**
Fixes wixtoolset/issues#6408

### CODE REVIEW

File: src/wix/test/WixToolsetTest.CoreIntegration/MsiFixture.cs [hunk 1]
```diff
@@ -873,7 +873,7 @@ public void FailsBuildAtLinkTimeForMissingEnsureTable()
                     first =>
                     {
                         Assert.Equal(MessageLevel.Error, first.Level);
-                        Assert.Equal("The identifier 'WixCustomTable:TableDefinitionNotExposedByExtension' could not be found. Ensure you have typed the reference correctly and that all the necessary inputs are provided to the linker.", first.ToString());
+                        Assert.Equal("Cannot find the table definitions for the 'TableDefinitionNotExposedByExtension' table.  This is likely due to a typing error or missing extension.  Please ensure all the necessary extensions are supplied on the command line with the -ext parameter.", first.ToString());
```
Comment (rseanhall) [inline src/wix/test/WixToolsetTest.CoreIntegration/MsiFixture.cs:876]: Please change the name of the test to match the new expected behavior. `FailsBuildAtBindTimeForMissingEnsureTable`?

**Review by rseanhall — COMMENTED** [review]
I would like to see a test that shows how to ensure an empty custom table that is defined in a Fragment instead of by an extension.

**Comment by rseanhall** [comment]
Also, need to verify Bundle container .cabs are the right size.

File: src/wix/WixToolset.Core.Burn/Bundles/ProcessMsiPackageCommand.cs [hunk 2]
```diff
@@ -397,6 +399,8 @@ private void ImportExternalCabinetAsPayloads(Database db, WixBundlePayloadSymbol
                                     Packaging = packagePayload.Packaging,
                                     ParentPackagePayloadRef = packagePayload.Id.Id,
                                 });
+
+                                this.CheckIfWindowsInstallerFileTooLarge(this.Facade.PackageSymbol.SourceLineNumbers, payloadSourceFile, "cabinet");
```
Comment (robmen) [inline src/wix/WixToolset.Core.Burn/Bundles/ProcessMsiPackageCommand.cs:403]: Doesn't this check if the cabinet files are too large in the bundle?

Comment (rseanhall) [reply src/wix/WixToolset.Core.Burn/Bundles/ProcessMsiPackageCommand.cs:403]: Yes, for `MsiPackage` files. But another check is needed for containers created by  https://github.com/wixtoolset/wix4/blob/73abf1db11a2ec38937682563d382a5c67fe2c40/src/wix/WixToolset.Core.Burn/Bundles/CreateContainerCommand.cs, right?

Comment (robmen) [reply src/wix/WixToolset.Core.Burn/Bundles/ProcessMsiPackageCommand.cs:403]: Not as long as Burn can handle the files. My understanding is that MSI can't handle cabinet files over 2GB not Burn.

Comment (rseanhall) [reply src/wix/WixToolset.Core.Burn/Bundles/ProcessMsiPackageCommand.cs:403]: I have not tested it. I guess it makes sense that if wixnative can create the cab then Burn should be able to read it, but I'm not 100% sure on that.

**Pull request merged**